src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/converter/rulemanager/OfFlowModConverter.java

```diff
@@ -0,0 +1,241 @@
+import org.projectfloodlight.openflow.types.TableId;
+import org.projectfloodlight.openflow.types.U64;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Mapper
+public abstract class OfFlowModConverter {
+    public static final OfFlowModConverter INSTANCE = Mappers.getMapper(OfFlowModConverter.class);
+
+
+    /**
+     * Convert stats reply.
+     */
+    public List<FlowSpeakerData> convertToFlowSpeakerCommandData(OFFlowStatsReply statsReply) {
+        List<FlowSpeakerData> flowSpeakerData = new ArrayList<>();
+        for (OFFlowStatsEntry entry : statsReply.getEntries()) {
+            FlowSpeakerData speakerData = FlowSpeakerData.builder()
+                    .cookie(new Cookie(entry.getCookie().getValue()))
+                    .flags(convertToRuleManagerFlags(entry.getFlags()))
+                    .priority(entry.getPriority())
+                    .table(OfTable.fromInt(entry.getTableId().getValue()))
+                    .match(OfMatchConverter.INSTANCE.convertToRuleManagerMatch(entry.getMatch()))
+                    .instructions(
+                            OfInstructionsConverter.INSTANCE.convertToRuleManagerInstructions(entry.getInstructions()))
+                    .build();
+            flowSpeakerData.add(speakerData);
+        }
+        return flowSpeakerData;
+    }
+
+    /**
+     * Convert stats reply.
+     */
+    public List<GroupSpeakerData> convertToGroupSpeakerCommandData(OFGroupDescStatsReply statsReply) {
+        List<GroupSpeakerData> commandData = new ArrayList<>();
+        for (OFGroupDescStatsEntry entry : statsReply.getEntries()) {
+        }
+        return commandData;
+    }
+
+    /**
+     * Convert group command data into OfGroupMod representation.
+     */
+    public OFGroupAdd convertInstallGroupCommand(GroupSpeakerData commandData, OFFactory ofFactory) {
+        List<OFBucket> buckets = commandData.getBuckets().stream()
+                .map(x -> toOfBucket(ofFactory, x)).collect(Collectors.toList());
+        return ofFactory.buildGroupAdd()
+                .setGroup(OFGroup.of((int) commandData.getGroupId().getValue()))
+                .setGroupType(toOfGroupType(commandData.getType()))
+                .setBuckets(buckets)
+                .build();
+
+    }
+
+    /**
+     * Convert group command data into OfGroupMod representation.
+     */
+    public OFGroupDelete convertDeleteGroupCommand(GroupSpeakerData commandData, OFFactory ofFactory) {
+        return ofFactory.buildGroupDelete()
+                .setGroup(OFGroup.of((int) commandData.getGroupId().getValue()))
+                .setGroupType(toOfGroupType(commandData.getType()))
+                .build();
+    }
+
+    private OFBucket toOfBucket(OFFactory ofFactory, Bucket bucket) {
+        List<OFAction> ofActions = OfInstructionsConverter.INSTANCE.convertApplyActions(bucket.getWriteActions(),
```

Comment:
Rename `convertApplyActions` -> `convertActions`. This method may be used for applyActions and writeActions.